src/overlay007/communication_club.c

```diff
@@ -0,0 +1,1731 @@
+        commClubMan->unk_00[v0] = Strbuf_Init((70 * 2), 4);
+    }
+}
+
+static void CommClubMan_Delete (void)
+{
+    int v0;
+
+    for (v0 = 0; v0 < 8; v0++) {
+        Strbuf_Free(commClubMan->unk_00[v0]);
+    }
+
+    Heap_FreeToHeap(commClubMan->unk_7C);
+
+    if (commClubMan->unk_50) {
+        sub_0200B3F0(commClubMan->unk_50);
+        commClubMan->unk_50 = NULL;
+    }
+
+    if (commClubMan->unk_54) {
+        sub_0200B3F0(commClubMan->unk_54);
+        commClubMan->unk_54 = NULL;
+    }
+
+    if (commClubMan->unk_58) {
+        sub_0200B3F0(commClubMan->unk_58);
+        commClubMan->unk_58 = NULL;
+    }
+
+    MessageLoader_Free(commClubMan->msgLoader);
+
+    if (commClubMan->unk_40.unk_0C != NULL) {
+        BGL_DeleteWindow(&commClubMan->unk_40);
+    }
+
+    Heap_FreeToHeap(commClubMan);
+    commClubMan = NULL;
+}
+
+static void CommClubMan_StartBattleClient (CommClubManager * man)
+{
+    FieldCommMan_StartBattleClient(man->fieldSystem, man->unk_91, CommClubMan_Regulation());
+}
+
+static const UnkStruct_ov84_02240FA8 Unk_ov7_0224ED34 = {
+    NULL,
+    ov7_02249C44,
+    ov7_02249C64,
+    NULL,
+    0x10,
+    0x5,
+    0x0,
+    0x8,
+    0x0,
+    0x0,
+    0x1,
+    0xF,
+    0x2,
+    0x0,
+    0x10,
+    0x1,
+    0x0,
+    0x0,
+    NULL
+};
+
+static void ov7_02249C44 (UnkStruct_0200112C * param0, u32 param1, u8 param2)
+{
+    commClubMan->unk_98 = 1;
+
+    if (param2 == 0) {
+        Sound_PlayEffect(1500);
+    }
+}
+
+static void ov7_02249C64 (UnkStruct_0200112C * param0, u32 param1, u8 param2)
+{
+    int v0;
+
+    for (v0 = 0; v0 < sub_02001504(param0, 3); v0++) {
+        ov7_02249C94(param0, 0, v0);
+    }
+}
+
+static void ov7_02249C94 (UnkStruct_0200112C * param0, u32 param1, u8 param2)
+{
+    int v0 = sub_02033808();
+    u16 v1 = 0;
+
+    if (commClubMan->unk_5C) {
+        sub_020014DC(commClubMan->unk_5C, &v1, NULL);
+    }
+
+    v1 += param2;
+
+    BGL_WindowColor(&commClubMan->unk_20, 15, 8, param2 * 16, sub_0201C294(&commClubMan->unk_20) * 8 - 8, 16);
+
+    if (v1 < v0) {
+        sub_020339AC(v1, commClubMan->unk_7C);
+        StringFormatter_FormatNumber(commClubMan->unk_50, 0, v1 + 1, 2, 2, 1);
+        StringFormatter_FormatPlayerName(commClubMan->unk_50, 1, commClubMan->unk_7C);
+        MessageLoader_GetStrbuf(commClubMan->msgLoader, 63, commClubMan->unk_00[1]);
+        StringFormatter_Format(commClubMan->unk_50, commClubMan->unk_00[0], commClubMan->unk_00[1]);
+        PrintStringSimple(&commClubMan->unk_20, 0, commClubMan->unk_00[0], 8, param2 * 16, 0xff, NULL);
+        StringFormatter_FormatNumber(commClubMan->unk_50, 2, TrainerInfo_ID(commClubMan->unk_7C) % 0x10000, 5, 2, 1);
+        MessageLoader_GetStrbuf(commClubMan->msgLoader, 65, commClubMan->unk_00[3]);
+        StringFormatter_Format(commClubMan->unk_50, commClubMan->unk_00[2], commClubMan->unk_00[3]);
+        PrintStringSimple(&commClubMan->unk_20, 0, commClubMan->unk_00[2], 11 * 8, param2 * 16, 0xff, NULL);
```

Comment:
I would break this into logical chunks:

```suggestion
        sub_020339AC(v1, commClubMan->unk_7C);

        StringFormatter_FormatNumber(commClubMan->unk_50, 0, v1 + 1, 2, 2, 1);
        StringFormatter_FormatPlayerName(commClubMan->unk_50, 1, commClubMan->unk_7C);
        MessageLoader_GetStrbuf(commClubMan->msgLoader, 63, commClubMan->unk_00[1]);
        StringFormatter_Format(commClubMan->unk_50, commClubMan->unk_00[0], commClubMan->unk_00[1]);
        PrintStringSimple(&commClubMan->unk_20, 0, commClubMan->unk_00[0], 8, param2 * 16, 0xff, NULL);

        StringFormatter_FormatNumber(commClubMan->unk_50, 2, TrainerInfo_ID(commClubMan->unk_7C) % 0x10000, 5, 2, 1);
        MessageLoader_GetStrbuf(commClubMan->msgLoader, 65, commClubMan->unk_00[3]);
        StringFormatter_Format(commClubMan->unk_50, commClubMan->unk_00[2], commClubMan->unk_00[3]);
        PrintStringSimple(&commClubMan->unk_20, 0, commClubMan->unk_00[2], 11 * 8, param2 * 16, 0xff, NULL);
```